server/server.go

```diff
@@ -562,62 +561,102 @@ func extractClientIDFromHost(hostName string) string {
 }
 
 func newRequest(
+	ctx context.Context,
 	clientIP net.IP, clientID string,
 	protocol model.RequestProtocol, request *dns.Msg,
-) *model.Request {
-	return &model.Request{
+) (context.Context, *model.Request) {
+	ctx, logger := log.CtxWithFields(ctx, logrus.Fields{
+		"req_id":    uuid.New().String(),
```

Comment:
@0xERR0R in #1307 you mentioned not using UUIDs, do you have something else in mind?  
I just went with that since it was already in `go.mod`, and unique across restarts, but changing should be trivial.

[xid](https://pkg.go.dev/github.com/rs/xid) might be a good option, and its docs has a small comparison that links alternatives.

Reply:
Looked into some UUID alternatives.  

I currently would prefer mid over most of em even if I never worked with it. 😅 
Reducing the string length by a lot sounds tempting.

Reply:
@0xERR0R any opinion on the ID format?

Reply:
We can use message id from dns request (dns.Id) itself, this is not to long. But UUID is also ok (I have only concerns about ID length, this could make log files difficult to read by human).

Reply:
DNS message ID is only unique per client so I think it's best to avoid it here.  
I'll just keep UUID as we already have it as a dependency and if it's annoying we can replace it easily.